lib/src/model.dart

```diff
@@ -649,6 +604,44 @@ class PkgDependency extends Object
   }
 }
 
+@JsonSerializable()
+class Health extends Object with _$HealthSerializerMixin {
+  /// The number of errors from `dartanalyzer`.
+  final int analyzerErrorCount;
+
+  /// The number of warnings from `dartanalyzer`.
+  final int analyzerWarningCount;
+
+  /// The number of hints/info messages from `dartanalyzer`.
+  final int analyzerHintCount;
+
+  /// The number of files having platform conflicts.
+  final int platformConflictCount;
+
+  /// The suggestions about the issues affecting the health score.
+  @JsonKey(includeIfNull: false)
+  final List<Suggestion> suggestions;
+
+  Health({
+    @required this.analyzerErrorCount,
+    @required this.analyzerWarningCount,
+    @required this.analyzerHintCount,
+    @required this.platformConflictCount,
+    @required List<Suggestion> suggestions,
+  }) : suggestions =
+            suggestions == null || suggestions.isEmpty ? null : suggestions;
+
+  factory Health.fromJson(Map<String, dynamic> json) => _$HealthFromJson(json);
+
+  double get healthScore {
+    double score = math.pow(0.75, analyzerErrorCount) *
+        math.pow(0.95, analyzerWarningCount) *
+        math.pow(0.995, analyzerHintCount);
+    score -= 0.25 * platformConflictCount;
```

Comment:
Not sure if logic like this belongs here, or whether it should be computed before this object is constructed, or by the thing using the object. Can this link to the relevant documentation?

Reply:
adding TODO for follow-up documentation